zooniverse/front-end-monorepo
# Add unread Messages count to ZooHeader, with sugar subscription

## Package
- app-project

## Linked Issue and/or Talk Post
- closes #2459 
- related to #3347 

## Describe your changes
- refactors existing unread Notifications count to `unreadNotificationsCount`
- adds initial request for unread Messages (aka Talk conversations), storing related ids in `unreadConversationsIds`, with `getTalkUnreadConversationsIds` helper
- adds additional logic for parsing sugar notifications related to messages (the subsets of conversations) so if the message is from a known unread conversation then unread Messages count is unchanged, but if the message is from a new unread conversation then unread Messages count is incremented

## How to Review

**SPECIAL NOTE**: I think we're going to have to review this functionality together, so we can:

1. create comments on discussions you subscribe to
2. reply to your comments
3. send you a message/conversation
4. other?

in real-time so you can see the appropriate counts increment. While testing, we'll need to change our [email settings](https://www.zooniverse.org/settings/email) to Never so notifications remain unread.

_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - https://www.zooniverse.org/projects/markbouslog/test-project-mb/talk
  - https://local.zooniverse.org:3000/projects/markbouslog/test-project-mb?env=production&demo=true  
- What user actions should my reviewer step through to review this PR? see 1-4 above
- Which storybook stories should be reviewed? N/A
- Are there plans for follow up PR’s to further fix this bug or develop this feature? no

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
